cpp/velox/substrait/SubstraitParser.cc

```diff
@@ -104,31 +104,41 @@ std::vector<TypePtr> SubstraitParser::parseNamedStruct(const ::substrait::NamedS
   return typeList;
 }
 
-std::vector<bool> SubstraitParser::parsePartitionColumns(const ::substrait::NamedStruct& namedStruct) {
+void SubstraitParser::parsePartitionAndMetadataColumns(
+    const ::substrait::NamedStruct& namedStruct,
+    std::vector<bool>& isPartitionColumns,
+    std::vector<bool>& isMetadataColumns) {
   const auto& columnsTypes = namedStruct.column_types();
-  std::vector<bool> isPartitionColumns;
   if (columnsTypes.size() == 0) {
-    // Regard all columns as non-partitioned columns.
+    // Regard all columns as regular columns.
```

Comment:
If `columnsTypes` has 0 as size, does it mean that there is no columns in this table? if so, maybe we should do a check `columnsTypes.size() > 0` instead?

Reply:
for read node it should be true, but not sure for write node if `namedStruct.names()` also is 0 size.

Reply:
@JkSelf will it happen for parquet write?

Reply:
so if is true for write then we only process under `columnsTypes.size() > 0` ?

Reply:
I think so but we'd better confirm first.